planet/clients/subscriptions.py

```diff
@@ -1,87 +1,89 @@
 """Planet Subscriptions API Python client."""
 
-import itertools
-from typing import AsyncIterator, Dict, Optional, Set
-import uuid
+import logging
+from typing import AsyncIterator, Optional, Set
 
-from planet.exceptions import APIError
+from httpx import URL
 
-# Collections of fake subscriptions and results for testing. Tests will
-# monkeypatch these attributes.
-_fake_subs: Dict[str, dict] = {}
-_fake_sub_results: Dict[str, list] = {}
+from planet.exceptions import APIError, ClientError, PagingError
+from planet.http import Session
+from planet.models import Paged
 
+LOGGER = logging.getLogger()
 
-async def _server_subscriptions_post(request):
-    missing_keys = {'name', 'delivery', 'source'} - request.keys()
-    if missing_keys:
-        raise RuntimeError(f"Request lacks required members: {missing_keys!r}")
 
-    id = str(uuid.uuid4())
-    _fake_subs[id] = request
-    sub = _fake_subs[id].copy()
-    sub.update(id=id)
-    return sub
+class _Pager(Paged):
+    """Navigates pages of messages about subscriptions or results."""
 
+    # This constructor overrides Paged's and takes an httpx
+    # async client as the first argument.
+    def __init__(self, client, request, limit=None):
+        self.request = request
+        self.client = client
+        self._pages = None
+        self._items = []
+        self.i = 0
+        self.limit = limit
```

Comment:
@jreiberkyle I subclassed Paged while developing this feature to reduce layers of abstraction and make things easier to debug. I do not mind at all if this extra class is deleted on the way to enabling automatic retries (if we need that).

I *do* think the Paged API would be cleaner if the constructor took a client (as I did) or session object as an argument instead of a "request doer". So maybe that could be something that comes out of this PR into other modules?